api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,271 @@
+	Snapshotter   *corev1.ResourceRequirements `json:"snapshotter,omitempty"`
+	Resizer       *corev1.ResourceRequirements `json:"resizer,omitempty"`
+	Provisioner   *corev1.ResourceRequirements `json:"provisioner,omitempty"`
+	OMapGenerator *corev1.ResourceRequirements `json:"omapGenerator,omitempty"`
+	Liveness      *corev1.ResourceRequirements `json:"liveness,omitempty"`
+	Plugin        *corev1.ResourceRequirements `json:"plugin,omitempty"`
+}
+
+type ProvisionerSpec struct {
+	// Embedded common pods spec
+	PodCommonSpec `json:"inline"`
+
+	// Set replicas for csi provisioner deployment. Defaults to 2
+	Replicas *int32 `json:"provisionerReplicas,omitempty"`
+
+	// Resource requirements for provisioner's containers
+	Resources ProvisionerResourcesSpec `json:"resources,omitempty"`
+}
+
+type LivenessSpec struct {
+	// Port to expose liveness metrics
+	MetricsPort int `json:"metricsPort,omitempty"`
+}
+
+type LeaderElectionSpec struct {
+	// Duration in seconds that non-leader candidates will wait to force acquire leadership.
+	// Default to 137 seconds.
+	LeaseDuration int `json:"leaseDuration,omitempty"`
+
+	// Deadline in seconds that the acting leader will retry refreshing leadership before giving up.
+	// Defaults to 107 seconds.
+	RenewDeadline int `json:"renewDeadline,omitempty"`
+
+	// Retry Period in seconds the LeaderElector clients should wait between tries of actions.
+	// Defaults to 26 seconds.
+	RetryPeriod int `json:"retryPeriod,omitempty"`
+}
+
+type CephFsClientType string
+
+const (
+	KernelCephFsClient CephFsClientType = "kernel"
+	FuseCephFsClient   CephFsClientType = "Fuse"
+)
+
+// DriverSpec defines the desired state of Driver
+type DriverSpec struct {
+	// Logging configuration for driver's pods
+	Logging *LogSpec `json:"logging,omitempty"`
+
+	// A reference to a ConfigMap resource holding image overwrite for deployed
+	// containers
+	ImageSet *corev1.LocalObjectReference `json:"imageSet"`
+
+	// Cluster name identifier to set as metadata on the CephFS subvolume and RBD images. This will be useful in cases
+	// like for example, when two container orchestrator clusters (Kubernetes/OCP) are using a single ceph cluster.
+	ClusterName *string `json:"clusterName,omitempty"`
+
+	// Set to true to enable adding volume metadata on the CephFS subvolume and RBD images.
+	// Not all users might be interested in getting volume/snapshot details as metadata on CephFS subvolume and RBD images.
+	// Hence enable metadata is false by default.
+	EnableMetadata *bool `json:"enableMetadata,omitempty"`
+
+	// Set the gRPC timeout for gRPC call issued by the driver components
+	GRpcTimeout int `json:"grpcTimeout,omitempty"`
+
+	// Set to true enable snapshot functionality (deplys a snapshooter sidecar)
+	// EnableVolumeSnapshoots     *bool `json:"enableVolumeSnapshoots,omitempty"`
+	// EnableVolumeGroupSnapshoot *bool `json:"enableVolumeGroupSnapshoot,omitempty"`
+	// none, autodetect, volume, volumeAndVolumeGroup
```

Comment:
`none, autodetect, volume, volumeAndVolumeGroup` to `none, autodetect, snapshotGroup, snapshot`

Reply:
👍

Reply:
As this is a snapshot-specific feature, it conveys the functionality the user is working with. Having a snapshot makes sense to me.

Reply:
ok, so done